src/Frontend/Core/Engine/Page.php

```diff
@@ -181,34 +191,43 @@ private function checkAuthentication(): void
 
     public function display(): Response
     {
-        // assign the id so we can use it as an option
-        $this->template->assignGlobal('isPage' . $this->pageId, true);
-        $this->template->assignGlobal('isChildOfPage' . $this->record['parent_id'], true);
+        try {
+            // assign the id so we can use it as an option
+            $this->template->assignGlobal('isPage' . $this->pageId, true);
+            $this->template->assignGlobal('isChildOfPage' . $this->record['parent_id'], true);
+
+            // hide the cookiebar from within the code to prevent flickering
+            $this->template->assignGlobal(
+                'cookieBarHide',
+                !$this->get('fork.settings')->get('Core', 'show_cookie_bar', false)
+                || $this->getContainer()->get('fork.cookie')->hasHiddenCookieBar()
+            );
 
-        // hide the cookiebar from within the code to prevent flickering
-        $this->template->assignGlobal(
-            'cookieBarHide',
-            !$this->get('fork.settings')->get('Core', 'show_cookie_bar', false)
-            || $this->getContainer()->get('fork.cookie')->hasHiddenCookieBar()
-        );
+            $this->parsePositions();
 
-        $this->parsePositions();
+            // assign empty positions
+            $unusedPositions = array_diff(
+                $this->record['template_data']['names'],
+                array_keys($this->record['positions'])
+            );
+            foreach ($unusedPositions as $position) {
+                $this->template->assign('position' . \SpoonFilter::ucfirst($position), []);
+            }
 
-        // assign empty positions
-        $unusedPositions = array_diff($this->record['template_data']['names'], array_keys($this->record['positions']));
-        foreach ($unusedPositions as $position) {
-            $this->template->assign('position' . \SpoonFilter::ucfirst($position), []);
-        }
+            $this->header->parse();
+            $this->breadcrumb->parse();
+            $this->parseLanguages();
+            $this->footer->parse();
 
-        $this->header->parse();
-        $this->breadcrumb->parse();
-        $this->parseLanguages();
-        $this->footer->parse();
+            return new Response(
+                $this->template->getContent($this->templatePath),
+                $this->statusCode
+            );
+        } catch (NotFoundHttpException $notFoundHttpException) {
+            $this->handlePage(Response::HTTP_NOT_FOUND);
```

Comment:
This could potentially cause a loop, if the 404-page is deleted. But #yolo

Reply:
the 404 cannot be deleted, it is protected

Reply:
I know, hence the #yolo